model/modeldecoder/span.go

```diff
@@ -58,15 +58,17 @@ func decodeSpan(input Input, schema *jsonschema.Schema) (transform.Transformable
 	fieldName := field.Mapper(input.Config.HasShortFieldNames)
 	decoder := utility.ManualDecoder{}
 	event := span.Event{
-		Metadata:      input.Metadata,
-		Name:          decoder.String(raw, fieldName("name")),
-		Start:         decoder.Float64Ptr(raw, fieldName("start")),
-		Duration:      decoder.Float64(raw, fieldName("duration")),
-		Sync:          decoder.BoolPtr(raw, fieldName("sync")),
-		Timestamp:     decoder.TimeEpochMicro(raw, fieldName("timestamp")),
-		Id:            decoder.String(raw, fieldName("id")),
-		ParentId:      decoder.String(raw, "parent_id"),
-		TraceId:       decoder.String(raw, "trace_id"),
+		Metadata:  input.Metadata,
+		Name:      decoder.String(raw, fieldName("name")),
+		Start:     decoder.Float64Ptr(raw, fieldName("start")),
+		Duration:  decoder.Float64(raw, fieldName("duration")),
+		Sync:      decoder.BoolPtr(raw, fieldName("sync")),
+		Timestamp: decoder.TimeEpochMicro(raw, fieldName("timestamp")),
+		Id:        decoder.String(raw, fieldName("id")),
+		ParentId:  decoder.StringPtr(raw, "parent_id"),
+		// ParentIdx comes from RUM V3 payloads only, and used to populate ParentId
+		ParentIdx:     decoder.IntPtr(raw, fieldName("parent_idx")),
```

Comment:
Can we just use "pi" here, rather than going through field mapping? It's only ever going to be relevant to RUMv3 isn't it?

Reply:
I considered it, but then I thought it would be convenient to have all the gibberish defined in the same place, so it works as a single-source-of truth (ie. documentation) for the rum agent folks. (the json-schema files are not a great documentation source anymore IMO)